building/configuring_git_workflow.adoc

```diff
@@ -0,0 +1,426 @@
+* 3852e7d - JBIDE-18372 - tweaking progress monitors (5 weeks ago, Rob Stryker)
+* 0d3f1a0 - JBIDE-18267 - ensure agent does not attach to newly detected jvms until
+* f61856b - (mine/JBIDE-17180_v2, JBIDE-17180_v2) JBIDE-17180 - label is too wide 
+```
+
+This is slightly easier to read, and is much preferred to normal merges. The reason this is acceptable is because:
+
+* It maintains the history of your merge branch, but
+* it separates it out visually, so users browsing history can skip over it entirely
+* It is still easy to read. 
+
+If you get into the bad habbit of performing many non-rebased merge commits, though, your git history will become 
+completely incomprehensible to any reader of the git commit log. With this in mind, 
+merge commits should be used very rarely and selectively. This is how such a messy history would appear to others:
+
+```
+*   4fec5c7 - (HEAD, git_test) Merge branch 'git_test4' into git_test (7 seconds ago, Rob Stryker)
+|\  
+| * dd39463 - (git_test4) Fixed a bug (21 seconds ago, Rob Stryker)
+| * db8ce15 - Added a data file v2 (32 seconds ago, Rob Stryker)
+| * 3d011ab - Added a data file (54 seconds ago, Rob Stryker)
+* |   44e2add - Merge branch 'git_test3' into git_test (2 minutes ago, Rob Stryker)
+|\ \  
+| * | 0099704 - (git_test3) Some critical pom stuff (2 minutes ago, Rob Stryker)
+* | |   9a2f565 - (mine/git_test) Merge pull request #13 from robstryker/git_test2 
+|\ \ \  
+| * | | 262b564 - (mine/git_test2, git_test2) Added a comment (12 minutes ago, Rob Stryker)
+| * | | 858a350 - Added a space (12 minutes ago, Rob Stryker)
+|/ / /  
+* | | 1108d9f - (origin/jbosstools-4.2.x, jbosstools-4.2.x) JBIDE-18886 - prepare for release
+* | | 006a285 - JBIDE-18737 - access to xpath model's map...
+* | | 47fbbcb - JBIDE-18513 - Remote server in mgmt mode...
+* | | 6cc6fef - JBIDE-18863 - browse button causes wrong...
+| |/  
+|/|   
+* | 3852e7d - JBIDE-18372 - tweaking progress monitors (5 weeks ago, Rob Stryker)
+|/  
+* 0d3f1a0 - JBIDE-18267 - ensure agent does not attach to newly...
+* f61856b - (mine/JBIDE-17180_v2, JBIDE-17180_v2) JBIDE-17180 
+```
+
+
+=== How to clean your local git repo
+
+Cleanliness involves two things:
+
+* ensuring you're up to date relative to the origin fork
+* purging old topic branches and pull-request branches, either because they're merged, or they're closed/discarded
+
+To stay current, you simply need to pull from your fork, and the origin fork, ensuring you merge in all upstream changes.
+
+You can ensure that you always merge and rebase by adding this to your `~/.gitconfig` file:
+
+  [branch]
+    autosetupmerge = true
+    autosetuprebase = always
+
+Then:
+
+    git checkout ${branch}
+    git pull origin ${branch}
+
+If you want to associate a local topic branch with a remote branch at the origin you can do this:
+
+    git checkout origin/${branch} -b ${topic}
+    git checkout ${topic}
```

Comment:
this is not about what I choose - it is about what git commands normally see in git docs and in daily usage.

I asked why you want to have two commands that must be executed together vs my suggestion that does the same but does not have this requirement of having to be executed together and can be used independently of each other. You seem to not have a reason for that (I actually thought you had one - which is why I asked), thus I suggest to use the one that has the least coupling and common use.